### PR TITLE
Document permissions around manual usage of Spack installation

### PR DESCRIPTION
**Pull request by nick-stroud**
Both of these recommendations (sudo -i and chained installs) have been tested against a running system.

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: community/modules/scripts/spack-setup/README.md [hunk 1]
```diff
@@ -119,6 +119,46 @@ sudo -i spack python -m pip install package-name
...
+    upstreams:
+      spack-instance-1:
+        install_tree: /sw/spack/opt/spack/
+    ```
+
+3. Add the following line to your `.bashrc` to make sure the new `spack` is in
+   your `PATH`.
+
+   ```sh
+   . <new_location>/spack/share/spack/setup-env.sh
```
Comment (tpdownes) [inline community/modules/scripts/spack-setup/README.md:159]: Has this been manually verified to work after the `/etc/profile.d` script activates the "upstream/system" Spack installation?

File: community/modules/scripts/spack-setup/README.md [hunk 2]
```diff
@@ -119,6 +119,46 @@ sudo -i spack python -m pip install package-name
...
+If there is a need to have a non-root user to install spack packages it is
+recommended to create a separate installation for that user and chain Spack installations
+([Spack docs](https://spack.readthedocs.io/en/latest/chain.html#chaining-spack-installations)).
+
+Steps to chaining Spack installations:
+
+1. Clone a new spack installation:
+
+    ```sh
+    git clone https://github.com/spack/spack.git <new_location>/spack
```
Comment (tpdownes) [inline community/modules/scripts/spack-setup/README.md:142]: nit: I would probably replicate the [spack clone instructions](https://github.com/spack/spack#-spack).

```
git clone -c feature.manyFiles=true https://github.com/spack/spack.git <new_location>/spack
```

File: community/modules/scripts/spack-setup/README.md [hunk 3]
```diff
@@ -119,6 +119,46 @@ sudo -i spack python -m pip install package-name
...
+
+Installed packages should be readable/executable by anyone on the VM.
+
+### Non-root package installation
+
+If there is a need to have a non-root user to install spack packages it is
+recommended to create a separate installation for that user and chain Spack installations
+([Spack docs](https://spack.readthedocs.io/en/latest/chain.html#chaining-spack-installations)).
+
+Steps to chaining Spack installations:
```
Comment (tpdownes) [inline community/modules/scripts/spack-setup/README.md:137]: "Steps to chain"

**Review by tpdownes — APPROVED** [review]
Minor change plus a nit. Feel free to squash/merge after fixing.

A quick search confirms that this upstream behavior appears to be the closest equivalent to conda's (much more natural) support for multi-user environments.

https://docs.conda.io/projects/conda/en/latest/user-guide/configuration/admin-multi-user-install.html

File: community/modules/scripts/spack-setup/README.md (hunk 1 above)
Comment (nick-stroud) [reply community/modules/scripts/spack-setup/README.md:159]: yes. I tested on spack installation using our spack-setup modules.

Comment (tpdownes) [reply community/modules/scripts/spack-setup/README.md:159]: I see that in description now (not in original notification)

File: community/modules/scripts/spack-setup/README.md (hunk 2 above)
Comment (nick-stroud) [reply community/modules/scripts/spack-setup/README.md]: Done

File: community/modules/scripts/spack-setup/README.md (hunk 3 above)
Comment (nick-stroud) [reply community/modules/scripts/spack-setup/README.md]: Done

**Pull request merged**